### PR TITLE
Switch to osbuild-composer backend for master

### PR DESCRIPTION
**Pull request by jkozol**
This is the same as PR #839 but to merged to master.

Per @larskarlitski 

>osbuild-composer is a drop-in replacement for lorax-composer, providing
the same API:
https://github.com/osbuild/osbuild-composer
Switch the dependency, documentation, and the name of the service for
the "Start Service" page.

### CODE REVIEW

**Comment by henrywang** [comment]
@jkozol, is osbuild-composer available on RHE-7 and Fedora 31/31? Thanks!

**Comment by jkozol** [comment]
@henrywang osbuild-composer is available on F31

**Comment by jkozol** [comment]
This PR should only land after https://github.com/cockpit-project/bots/pull/399 and #871

**Comment by henrywang** [comment]
@jkozol, the following fix should work for you.
```diff
diff --git i/test/end-to-end/wdio.conf.js w/test/end-to-end/wdio.conf.js
index f43e2c6e..3f8166d2 100644
--- i/test/end-to-end/wdio.conf.js
+++ w/test/end-to-end/wdio.conf.js
@@ -253,7 +253,7 @@ exports.config = {
     browser.login();
     browser.switchToComposerFrame();
     // only the first suite needs start lorax-composer
-    if (suite.title === "lorax-composer api sanity test") {
+    if (suite.title === "weldr api sanity test") {
       browser.startLoraxIfItDoesNotStart();
     }
   },
```

**Comment by henrywang** [comment]
@jkozol, I fixed imageBuilding test according to new osbuild-composer API. This PR is ready to go, I think. Could you please fix the conflict issue? I can't do that. Thanks!

**Comment by henrywang** [comment]
@jkozol, I'm wondering why the `should show correct download API in download link` case in imageBuilding.test.js got passed? Did you add `/api/v0/compose/status/*?blueprint=${name}` API support in osbuild-composer? As far as I found in the source code, only `uuid` and `*` can be supported. Did you see my push to your branch? I fixed that with `/api/v0/compose/status/*` API. Thanks! :+1:

**Comment by jkozol** [comment]
@henrywang I added filtering in this PR https://github.com/osbuild/osbuild-composer/pull/229. We can now filter by `blueprint`, `status`, and `type`.

**Comment by henrywang** [comment]
@jkozol COOL!

File: components/EmptyState/EmptyStateInactive.js [hunk 1]
```diff
@@ -57,14 +57,14 @@ class EmptyStateInactive extends React.Component {
 
   startService(e) {
     if (!e || e.button !== 0) return;
-    let argv;
+    let script;
     if (this.state.enableService) {
-      argv = ["systemctl", "enable", "--now", "osbuild-composer.socket"];
+      script = "systemctl enable --now osbuild-composer.socket && systemctl enable --now osbuild-worker@1";
```
Comment (ondrejbudai) [inline components/EmptyState/EmptyStateInactive.js]: Is it really needed to `enable --now osbuild-worker@1`? Doesn't it mean that osbuild-worker is always running in the background even when osbuild-composer isn't running?

Comment (jkozol) [reply components/EmptyState/EmptyStateInactive.js]: You are correct. We only need to start the socket. Given David's comment here https://github.com/osbuild/osbuild-composer/pull/330#discussion_r393570358 I am now wondering if we should enable the socket or just start it.

**Comment by henrywang** [comment]
@jkozol, I made some update to make Edge test more stable. Could you please update `wdio.conf.js`? Thanks!
```diff
diff --git i/test/end-to-end/wdio.conf.js w/test/end-to-end/wdio.conf.js
index e115f351..ff4ab51d 100644
--- i/test/end-to-end/wdio.conf.js
+++ w/test/end-to-end/wdio.conf.js
@@ -230,6 +230,13 @@ exports.config = {
     // wait a second here to make Edge happy
     browser.pause(1000);
 
+    // make browser window maximum
+    if (browser.capabilities.browserName.toLowerCase().includes("edge")) {
+      browser.execute(() => window.resizeTo(1280, 1024));
+    } else {
+      browser.setWindowSize(1280, 1024);
+    }
+
     // Add three pre-defined blueprints
     browser.login();
     browser.switchToComposerFrame();
```

**Comment by henrywang** [comment]
@jkozol, Can we land this PR firs? I'll send another PR to resolve flaky issue. The flaky issue on Edge is related with osbuild-composer, so I need land this PR first. Thanks!

**Comment by jkozol** [comment]
@henrywang I am happy for this to land now since chrome and firefox pass consistently. Can you approve it?

**Pull request merged**